docs/learn/guides/file-integrations/file-integration-using-ftp/README.md

```diff
@@ -0,0 +1,198 @@
+    string filePath;
+};
+
+Config conf = {
+    fileNamePattern: config:getAsString("FTP_FILE_NAME_PATTERN"),
+    filePath: config:getAsString("FTP_LISTENER_PATH")
+};
+```
+
+Create an FTP listener instance by defining the configuration.
+
+```ballerina
+listener ftp:Listener dataFileListener = new({
+    protocol: ftp:FTP,
+    host: config:getAsString("FTP_HOST"),
+    port: config:getAsInt("FTP_LISTENER_PORT"),
+    secureSocket: {
+        basicAuth: {
+            username: config:getAsString("FTP_USERNAME"),
+            password: config:getAsString("FTP_PASSWORD")
+        }
+    },
+    path: conf.filePath,
+    fileNamePattern: conf.fileNamePattern
+    pollingInterval: config:getAsInt("FTP_POLLING_INTERVAL"),
+});
+```
+
+Then implement the FTP client endpoint to do the file processing.
+     
+```ballerina
+ftp:ClientEndpointConfig ftpConfig = {
+    protocol: ftp:FTP,
+    host: config:getAsString("FTP_HOST"),
+    port: config:getAsInt("FTP_LISTENER_PORT"),
+    secureSocket: {
+     basicAuth: {
+         username: config:getAsString("FTP_USERNAME"),
+         password: config:getAsString("FTP_PASSWORD")
+     }
+    }
+};
+
+ftp:Client ftpClient = new(ftpConfig);
+```
+
+Create the service to listen to the FTP server. 
+
+```ballerina
+service dataFileService on dataFileListener {
+    resource function processDataFile(ftp:WatchEvent fileEvent) {
+
+        foreach ftp:FileInfo file in fileEvent.addedFiles {
+            log:printInfo("Added file path: " + file.path);
+            processNewFile(file.path);
+        }
+        foreach string file in fileEvent.deletedFiles {
+            log:printInfo("Deleted file path: " + file);
+            processDeletedFile(file.path);
+        }
+    }
+}
+```
+
+Declare a map to store file names and sizes.
+
+```ballerina
+map<int> fileMap = {};
+```
+
+Now, implement the processing of added and deleted files.
+
+```ballerina
+function processNewFile(string filePath) {
+    int|error fileSize = ftpClient -> size(filePath);
+    if(fileSize is int){
+        fileMap[filePath] = response;
+        log:printInfo("Added file: " + filePath + " - " + fileSize.toString());
+    } else {
+        log:printError("Error in getting file size", response);
+    }
+}
+
+function processDeletedFile(string filePath) {
+    if(fileMap.hasKey(filePath)){
+        fileMap.remove(filePath);
+        log:printInfo("Deleted file: " + filePath);
+    }
+}
+```
+
+## Testing
+
+### Invoking the service
+
+Start the FTP server you will listen to before starting the listener service.
+
+Navigate to `file-integration-using-ftp` and run the following command to start the listener service in `ftp_listener.bal`.
+
+```bash
+   $ ballerina run ftp_listener
+```
+
+Add and delete files in the FTP server and check the logs to verify the service is working as expected.
+
+## Deployment
+
+After the development process, you can deploy the services using below methods by selecting as you wish.
+
+### Deploying locally
```

Comment:
Do you need to run any `java jar` command as well?

Reply:
Added.